src/TraceEvent/Symbols/SymbolReader.cs

```diff
@@ -2681,16 +2681,16 @@ unsafe internal SourceFile(SymbolModule module, IDiaSourceFile sourceFile)
             // 1 CALG_MD5 checksum generated with the MD5 hashing algorithm.
             // 2 CALG_SHA1 checksum generated with the SHA1 hashing algorithm.
             m_hashType = sourceFile.checksumType;
-            if (m_hashType != 2 && m_hashType != 1 && m_hashType != 0)
-            {
-                Debug.Assert(false, "Unknown hash type");
-                m_hashType = 0;
-            }
             if (m_hashType != 0)
             {
+                uint hashSizeInBytes;
+                fixed (byte* bufferPtr = m_hash)
+                    sourceFile.get_checksum(0, out hashSizeInBytes, out *bufferPtr);
```

Comment:
Passing zero causes the API to tell you how big the hash needs to be. I'm not sure the syntax is optimal, though: in C++, you pass NULL for the final argument; I don't know what the equivalent is in managed code.